### PR TITLE
Small fixes for requests package

### PR DESCRIPTION
**Pull request by maciejaszek**

### CODE REVIEW

File: requests/requests.go
Comment (aalexanderr) [inline requests/requests.go:388]: why limit read access for other goroutines?

Comment (maciejaszek) [reply requests/requests.go:388]: Because in ProlongAccess Job is modified.

Comment (aalexanderr) [reply requests/requests.go:388]: is it to ease maintainability as potential performance increase is neglible while this was the only func in requests.go that modified req while allowing other goroutines to read it?

**Review by aalexanderr — APPROVED** [review]

File: requests/requests.go
Comment (maciejaszek) [reply requests/requests.go:388]: I think that was done rather by mistake than on purpose.

**Review by pawiecz — APPROVED** [review]

**Comment by pawiecz** [comment]
What is your preferred way of handling changes to the code base (self-merge after getting approval, final approver merges, etc.)? I'm fine with any, I'm just not 100% sure if you'd like me to do it or if you're taking care of it yourself.

**Comment by maciejaszek** [comment]
I think we should stick to the process that we used on gerrithub, i.e. maintainer merges (in case of Boruta that'll be me). In case of maintainer PRs author should wait for review of others. In case of any controversies/doubts (like in ProlongAccess in this PR) I'll wait for more opinions.

**Pull request merged**
